litellm/utils.py

```diff
@@ -1090,14 +1090,20 @@ def pre_call(self, input, api_key, model=None, additional_args={}):
             )
 
             verbose_logger.debug(f"PRE-API-CALL ADDITIONAL ARGS: {additional_args}")
-
-            curl_command = "\n\nPOST Request Sent from LiteLLM:\n"
-            curl_command += "curl -X POST \\\n"
-            curl_command += f"{api_base} \\\n"
-            curl_command += (
-                f"{formatted_headers} \\\n" if formatted_headers.strip() != "" else ""
-            )
-            curl_command += f"-d '{str(data)}'\n"
+            
```

Comment:
this looks good - but it also means we'll be doing a `json.dumps` per call -> which significantly impacts latency 

can you utilize the value of ` verbose_logger.level` or `litellm.set_verbose` and conditionally run this logic 

```
 formatted_headers_str = "{formatted_headers} \\\n".format(formatted_headers=formatted_headers) if formatted_headers.strip() != "" else ""
            try:
              data_str = json.dumps(data, default=str)
            except:
              data_str = str(data)

            curl_command = (
            "\n\nPOST Request Sent from LiteLLM:\n"
            "curl -X POST \\\n"
            "{api_base} \\\n"
            "{formatted_headers_str}-d '{data_str}'\n"
            ).format(api_base=api_base, formatted_headers_str=formatted_headers_str, data_str=data_str.replace("'", "'\"'\"'"))
```

Reply:
Done.